src/Microsoft.AspNetCore.Razor.Evolution/DefaultDocumentClassifierPass.cs

```diff
@@ -15,5 +16,35 @@ protected override bool IsMatch(RazorCodeDocument codeDocument, DocumentIRNode i
         {
             return true;
         }
+
+        protected override void OnDocumentStructureCreated(
+            RazorCodeDocument codeDocument,
+            NamespaceDeclarationIRNode @namespace,
+            ClassDeclarationIRNode @class,
+            RazorMethodDeclarationIRNode method)
+        {
+            base.OnDocumentStructureCreated(codeDocument, @namespace, @class, method);
+            var configuration = Engine.Features.OfType<RazorEngineConfigurationFeature>().FirstOrDefault();
+            if (configuration != null)
+            {
+                for (var i = 0; i < configuration.ConfigureClass.Count; i++)
+                {
+                    var configureClass = configuration.ConfigureClass[i];
+                    configureClass(@class);
```

Comment:
It would be nice to pass in the code document here, this way you can make a decision based on what you're looking at.